src/FinalForm.js

```diff
@@ -814,7 +814,7 @@ function createForm<FormValues: FormValuesShape>(
             fieldConfig.initialValue
           )
         }
-        if (fieldConfig.defaultValue !== undefined) {
+        if (fieldConfig.defaultValue !== undefined && fieldConfig.initialValue === undefined && (state.formState.initialValues || {})[name] === undefined) {
```

Comment:
What about for fields with a name like `asdf.zcxv`? Maybe it should use `getIn` instead, something more like this:
```suggestion
        if (fieldConfig.defaultValue !== undefined && fieldConfig.initialValue === undefined && getIn(state.formState.initialValues || {}, name) === undefined) {
```